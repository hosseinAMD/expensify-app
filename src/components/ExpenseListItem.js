import React from 'react';
import {Link} from 'react-router-dom';
import moment from 'moment';

const ExpenseListItem = (props) => (
    <div>
        <Link to={`/edit/${props.expense.id}`}><h3>{props.expense.description}</h3></Link>
        <p>{`${props.expense.amount} Rials`}
        --
        {moment(props.expense.createdAt).format('MMMM Do, YYYY')}</p>
    </div>
);

export default ExpenseListItem;