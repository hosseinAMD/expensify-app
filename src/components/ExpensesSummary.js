import React from 'react';
import {connect} from 'react-redux';
import selectExpenses from '../selectors/expenses';
import selectExpensesTotal from '../selectors/expense-total';
import numeral from 'numeral';

export const ExpensesSummary = (props) => (
    <div>
        <h3>{`Viewing ${props.expensesCount} expenses totalling ${numeral(props.expensesTotal).format('0,0')} Rials`}</h3>
    </div>
);

const mapStateToProps = (state) => {
    const visibleExpenses = selectExpenses(state.expenses,state.filters);
    return {
        expensesCount: visibleExpenses.length,
        expensesTotal: selectExpensesTotal(visibleExpenses)
    };
};

export default connect(mapStateToProps)(ExpensesSummary);