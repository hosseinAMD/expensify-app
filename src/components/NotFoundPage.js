import React from 'react';
import {Link} from 'react-router-dom';

const NotFoundPage = () => (
    <p>404 Page!!! - <Link to="/edit/53">Go Home!</Link></p>
);

export default NotFoundPage;