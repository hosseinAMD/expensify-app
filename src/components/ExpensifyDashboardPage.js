import React from 'react';
import ExpenseList from './ExpenseList';
import ExpenseListFilters from './ExpenseListFilters';
import ExpensesSummary from './ExpensesSummary';

const ExpensifyDashboardPage = () => (
    <div>
        <p>Dashboard Page</p>
        <ExpensesSummary />
        <ExpenseListFilters />
        <ExpenseList />
    </div>
);

export default ExpensifyDashboardPage;