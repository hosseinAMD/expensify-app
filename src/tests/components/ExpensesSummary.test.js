import React from 'react';
import {shallow} from 'enzyme';
import {ExpensesSummary} from '../../components/ExpensesSummary';


test('should render expenses summary correctly', () => {
    const wrapper = shallow(<ExpensesSummary expensesCount={2} expensesTotal={30000} />);
    expect(wrapper).toMatchSnapshot();
});