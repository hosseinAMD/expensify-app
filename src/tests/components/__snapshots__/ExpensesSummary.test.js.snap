// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`should render expenses summary correctly 1`] = `
<div>
  <h3>
    Viewing 2 expenses totalling 30,000 Rials
  </h3>
</div>
`;
